### PR TITLE
#165661809 Enhance get articles route

### PR DESCRIPTION
**Pull request by valenstical**
#### What does this PR do?
This PR modifies the GET /articles route to fetch articles based on authorId and include the total number of comments to search results.

#### Description of Task to be completed?

- [x] Write integration tests
- [x] Remove limit on pagination size 
- [x] Handle validation from middleware
- [x] Add functionality to get articles by authorId
- [x] Include the total number of comments to results returned by getArticle

#### How should this be manually tested?

1. Run the following commands on the terminal:

```bash
git clone -b bug/165661809/article-stats --single-branch \
https://github.com/andela/apollo-ah-backend.git

cd apollo-ah-backend
cp .env.example .env

# To run test
cp .env.example .env.test

# Install package modules
npm install
```

2. Modify `.env.test` file with your db credentials

3. Run database migrations with `npm run migrate`

4. Run seeders with `npm run seed`

5. Run `npm run start:dev` to spin up the server.

#### Running test locally

Two npm scripts are available to run the test suite:

- `npm run mocha` or `npm run mocha:watch` - The later watches for any file changes and runs the full test suite (without code coverage)
- `npm test` - Performs a single full test suite run, including Istanbul code coverage reporting. Summary coverage reports are written to stdout, and detailed HTML reports are available in /coverage/index.html

**NOTE:** To run test locally, make sure to create `.env.test` file.

#### What are the relevant pivotal tracker stories?

[#165661809](https://www.pivotaltracker.com/story/show/165661809)

#### Screenshots (if appropriate)
N/A

### CODE REVIEW

File: server/helpers/nestedDataProvider.js [hunk 1]
```diff
@@ -33,12 +33,24 @@ const ratingsModelObj = {
...
+  as: 'comments',
+};
+
 /**
  *
  *
  * @param {*} categoryQuery - category Id needed to query the database
  * @param {*} authorQuery - author name needed to query the DB
  * @param {*} tagQuery - tag name needed to query the DB
+ * @param {*} clapsQuery - tag name needed to query the DB
```
Comment (thislekan) [inline server/helpers/nestedDataProvider.js]: this still has the description for `tagQuery`

**Review by thislekan — CHANGES_REQUESTED** [review]
@valenstical well done, Valentine. Please attend to my requested changes.

**Review by aanchirinah — APPROVED** [review]
LGTM

File: server/helpers/nestedDataProvider.js (hunk 1 above)
Comment (valenstical) [reply server/helpers/nestedDataProvider.js]: Done. Thanks for pointing it out.

**Review by thislekan — APPROVED** [review]
LGTM

**Pull request merged**